### DISCUSSION TITLE
Is there something like `"UpstreamScheme":"wss"`?

### DISCUSSION THREAD

**Discussion by tp1415926535** [ThreeMammals/Ocelot#1965] [Q&A] [answered]
I'm using websocket, but I can only access the http port through ws, I can't connect to the https port of wss.
And there doesn't seem to be an '_UpstreamScheme'_ configuration similar to 'DownstreamScheme'. I tried that. It didn't work.

**Comment by raman-m** [comment, marked as answer]
- [WebSocket Secure](https://ocelot.readthedocs.io/en/latest/features/websockets.html#websocket-secure)

Read it carefully!

----
> And there doesn't seem to be an `UpstreamScheme` configuration similar to `DownstreamScheme`.

Sure, there is no such property in [Global Configuration](https://ocelot.readthedocs.io/en/latest/features/configuration.html) or [Routing](https://ocelot.readthedocs.io/en/latest/features/routing.html)!
Upstream scheme (protocol) is managed by your Ocelot app hosting config. See [GlobalConfiguration.BaseUrl](https://ocelot.readthedocs.io/en/latest/introduction/gettingstarted.html#configuration) property.
You can host Ocelot app as any ASP.NET app using `https` scheme only. If you would like to host using `wss` then I'm not sure what will happen. We don't support your upstream scenario. It could be a new feature.
But we support `https` to `wss` direction. It should work.

**Reply by tp1415926535** [reply]
Yes I found that just because I specified Urls for asp.net it was causing the other ports to fail, I removed that configuration and had no problems.
